src/FieldEncryptProcessEntities.php

```diff
@@ -101,26 +135,72 @@ public function decryptEntity(ContentEntityInterface $entity) {
    * @return string
    *   The processed value.
    */
-  protected function processValue($value = '', EncryptionProfileInterface $encryption_profile, $op = 'encrypt') {
+  protected function processValue(ContentEntityInterface $entity, FieldItemListInterface $field, $property_name, $value = '', EncryptionProfileInterface $encryption_profile, $op = 'encrypt') {
     // Do not modify empty strings.
     if ($value === ''){
       return '';
     }
 
     if ($op === 'encrypt') {
-      return base64_encode($this->encryptService->encrypt($value, $encryption_profile));
+      // Encrypt property value.
+      $processed_value = base64_encode($this->encryptService->encrypt($value, $encryption_profile));
+      // Save encrypted value in EncryptedFieldValue entity.
+      $this->encryptedFieldValueManager->saveEncryptedFieldValue($entity, $field->getName(), $property_name, $processed_value);
+      // Return value to store for unencrypted property.
+      // We can't set this to NULL, because then the field values are not saved,
+      // so we can replace them with their unencrypted value on load.
+      return '[ENCRYPTED]';
+
     }
     elseif ($op === 'decrypt') {
-      return $this->encryptService->decrypt(base64_decode($value), $encryption_profile);
+      // Get encrypted value from EncryptedFieldValue entity.
+      if ($encrypted_value = $this->encryptedFieldValueManager->getEncryptedFieldValue($entity, $field->getName(), $property_name)) {
+        // Decrypt value.
+        $decrypted_value = $this->encryptService->decrypt(base64_decode($encrypted_value), $encryption_profile);
+        return $decrypted_value;
+      }
+      else {
+        return $value;
+      }
     }
-    else {
-      return '';
+  }
+
+  /**
+   * @param \Drupal\Core\Field\FieldItemListInterface $field
+   *   The field to check.
+   * @return bool
+   */
+  protected function checkField(FieldItemListInterface $field) {
+    if (!is_callable([$field, 'getFieldDefinition'])){
```

Comment:
Wow, would be cool to have some comment here. I didn't realize this would even need to be done! :)